presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -263,6 +263,81 @@ public void testSemiJoinDynamicFilteringBlockProbeSide()
                 ImmutableSet.of(1, ORDERS_COUNT, PART_COUNT));
     }
 
+    @Test
+    public void testCrossJoinDynamicFiltering()
+    {
+        // "<" pushdown
+        {
+            assertDynamicFiltering(
+                    "SELECT * FROM lineitem JOIN orders ON " +
+                            "lineitem.orderkey < orders.orderkey + 1 AND " +
+                            "orders.comment = 'nstructions sleep furiously among '",
+                    withBroadcastJoin(),
+                    6,
+                    ImmutableSet.of(6, ORDERS_COUNT));
+        }
+        // "<=" pushdown
+        {
+            assertDynamicFiltering(
+                    "SELECT * FROM lineitem JOIN orders ON " +
+                            "lineitem.orderkey <= orders.orderkey AND " +
+                            "orders.comment = 'nstructions sleep furiously among '",
+                    withBroadcastJoin(),
+                    6,
+                    ImmutableSet.of(6, ORDERS_COUNT));
+        }
+        // "<=" and ">" pushdown
+        {
+            assertDynamicFiltering(
+                    "SELECT * FROM lineitem JOIN orders ON " +
+                            "lineitem.orderkey <= orders.orderkey AND " +
+                            "lineitem.orderkey > orders.orderkey - 1 AND " +
```

Comment:
Could we choose range (or remove comment predicate) so that number of results is less than 6 when clamped (too show that both comparisons are effective)

`lineitem.orderkey <= orders.orderkey` alone ensures 6 rows

Reply:
I will rewrite the tests to verify this scenario.